### PR TITLE
configure yamllint pre-commit step

### PR DESCRIPTION
**Pull request by joeyorlando**
# What this PR does

Add [`yamllint`](https://github.com/adrienverge/yamllint) to `pre-commit` configuration + fix pre-existing errors

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

**Review by jdbaldry — APPROVED** [review]
I think in theory these changes to the workflows I own look fine.
It's probably worth just keeping this PR in mind the next time docs are updated.

Out of interest, do you understand why `yamllint` is against the unquoted key `on`? It doesn't seem to mind any other keys haha.

File: .github/workflows/helm_release.yml
[the remark this answers is not in the document]
Comment (joeyorlando) [reply .github/workflows/helm_release.yml:1]: @jdbaldry you were asking about this, I found this odd at first as well

**Comment by joeyorlando** [comment]
[this SO answer](https://stackoverflow.com/a/21699210) gives a very good summary on the (too many) ways there are to write multi-line yaml strings, and their syntaxes

File: .github/workflows/helm_release.yml
[the remark this answers is not in the document]
Comment (joeyorlando) [reply .github/workflows/helm_release.yml:1]: TIL about the "Norway problem".. woww 😆

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -103,7 +95,16 @@ repos:
...
         name: markdownlint - docs/sources
         entry: markdownlint --fix --ignore README.md -c ./docs/.markdownlint.json ./docs/sources/**/*.md
+
+  - repo: https://github.com/adrienverge/yamllint
+    rev: v1.32.0
+    hooks:
+      - id: yamllint
+        name: yamllint
+        entry: yamllint -c .yamllint.yml
+        types: [yaml]
```
Comment (joeyorlando) [inline .pre-commit-config.yaml:110]: main change in this PR, rest is fixing pre-existing `yamllint` errors

File: .yamllint.yml
Comment (joeyorlando) [inline .yamllint.yml:1]: main change in this PR, rest is fixing pre-existing `yamllint` errors

**Pull request merged**